src/Core/src/Handlers/View/ViewHandler.Windows.cs

```diff
@@ -1,31 +1,35 @@
 ﻿#nullable enable
 using System;
-using System.Collections.Generic;
-using System.ComponentModel;
+using System.Runtime.CompilerServices;
 using Microsoft.UI.Xaml;
-using Microsoft.UI.Xaml.Controls;
 using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml.Input;
 using PlatformView = Microsoft.UI.Xaml.FrameworkElement;
 
 namespace Microsoft.Maui.Handlers
 {
 	public partial class ViewHandler
 	{
+		readonly static ConditionalWeakTable<PlatformView, ViewHandler> FocusManagerMapping = new();
+
+		static ViewHandler()
+		{
+			FocusManager.GotFocus += FocusManager_GotFocus;
+			FocusManager.LostFocus += FocusManager_LostFocus;
+		}
+
 		partial void ConnectingHandler(PlatformView? platformView)
 		{
-			if (platformView != null)
+			if (platformView is not null)
 			{
-				platformView.GotFocus += OnPlatformViewGotFocus;
-				platformView.LostFocus += OnPlatformViewLostFocus;
+				FocusManagerMapping.Add(platformView, this);
```

Comment:
Is there no perf issues when there is a super complex UI? If we have maybe 1k elements on screen, will not the add and lookup be slow? It may still be faster than what we had before, but I am getting flashbacks when millions of objects would sub and unsub from the theme changed event.

Reply:
> when millions of objects would sub and unsub from the theme changed event

Removing an entry from a CWT seems very fast compared to event unsubscription.

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConditionalWeakTable.cs#L614

I would rather set an initial capacity of 128 to the CWT to avoid a lot of initial resize operation (which is expensive).

If you're still concerned we have an alternative but much slower implementation:
https://github.com/dotnet/maui/pull/24355